mobile/src/components/course/components/edit-course-category/edit-course-category.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { FC } from 'react';
+
+import { AppScreenName, DataStatus } from '~/common/enums/enums';
+import {
+  CategoryGetAllItemResponseDto,
+  CourseGetResponseDto,
+} from '~/common/types/types';
+import {
+  BackButton,
+  Button,
+  Dropdown,
+  Spinner,
+  View,
+} from '~/components/common/common';
+import {
+  useAppDispatch,
+  useAppForm,
+  useAppNavigate,
+  useAppSelector,
+  useCallback,
+  useEffect,
+  useFocusEffect,
+} from '~/hooks/hooks';
+import { coursesActions } from '~/store/actions';
+import { courseUpdateCategory as courseUpdateCategoryValidationSchema } from '~/validation-schemas/validation-schemas';
+
+import { DropDownPayload } from './common/common';
+import { getDefaultValueforDropDown } from './helper/helpers';
+import { styles } from './styles';
+
+const EditCourseCategory: FC = () => {
+  const navigation = useAppNavigate();
+  const dispatch = useAppDispatch();
+
+  const { course, dataStatus, categories } = useAppSelector(
+    (state) => state.courses,
+  );
+  const currentCategory = course?.courseCategoryId;
+
+  const { control, handleSubmit, reset, errors } = useAppForm<
+    DropDownPayload | CategoryGetAllItemResponseDto
```

Comment:
after that, I have errors from TS. Because categories has type CategoryGetAllItemResponseDto[] . And I am confused.
![image](https://user-images.githubusercontent.com/82529236/187185021-fc76fd2a-90ec-4b27-b407-27db4f1ed150.png)

Reply:
Your items type in select should be `ItemType<string>[]`

Reply:
Then have problems with items type. Don't understand how works, in that case TS, going to documentation

![image](https://user-images.githubusercontent.com/82529236/187191087-2011cd2c-1b2a-4fad-8de0-f12a8ba5513d.png)

Reply:
It's because you items has incorrect interface, they should look like array with label and value properties. You should convert categories to it

Reply:
Can I use for convert categories  something like that:
![image](https://user-images.githubusercontent.com/82529236/187209104-7c882d5d-afd9-4112-8d03-82fae47795de.png)